sourceinfo.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protoschema
+
+import (
+	"embed"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bufbuild/protoschema-plugins/internal/protoschema/plugin/pluginsourceinfo"
+	"github.com/jhump/protoreflect/desc/sourceinfo"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+var (
+	// SourceInfoGlobalTypes is a replacement for protoregistry.GlobalTypes that includes
+	// all registered source info.
+	SourceInfoGlobalTypes = sourceinfo.GlobalTypes
+	// SourceInfoGlobalFiles is a replacement for protoregistry.GlobalFiles that includes
+	// all registered source info.
+	SourceInfoGlobalFiles = sourceinfo.GlobalFiles
```

Comment:
As declared, these leak types from the `sourceinfo` package into the exported API of this package. Also, the type of `sourceinfo.GlobalFiles` is icky (there are comments in the code; it's only that way for back-compat, since that module is post-v1.0).

How about instead:
```
SourceInfoGlobalFiles protodesc.Resolver = sourceinfo.GlobalFiles

SourceInfoGlobalTypes interface {
	protoregistry.MessageTypeResolver
	protoregistry.ExtensionTypeResolver
	RangeExtensionsByMessage(message protoreflect.FullName, f func(protoreflect.ExtensionType) bool)
} = sourceinfo.GlobalTypes
```

Also, these seem awkward to be exported API in the `protoschema` package, since they seem rather specialized and not directly related to manipulating protobuf schemas and schema formats. Maybe they should be moved into a sub-package?

Reply:
done